### ISSUE TITLE
Make the ASP.NET Core MVC scaffolders use Bootstrap 4

### ISSUE THREAD

**Issue by TheMegaMatt** [aspnet/Scaffolding#712]
or at least switch some way between the two, especially for the CRUD scaffolding, makes the feature useless, its more work to fix the one what got generated that writing from scratch

**Comment by danroth27** [comment]
We will need to support both Bootstrap 3 and 4 for folks that are migrating apps that use Bootstrap 3.

**Comment by danroth27** [comment]
Updating the title since the template work is tracked by https://github.com/aspnet/templating/issues/278

**Comment by TheMegaMatt** [comment]
Would it be compete madness having a switch to have 3, 4 or none? Like
having a starter template but with just some really basic html so people
can chose what framework to use

Il ven 22 giu 2018, 18:37 Daniel Roth <notifications@github.com> ha scritto:

**Comment by danroth27** [comment]
@TheMegaMatt Sounds like a reasonable ask, but please file a separate issue so that we can track the ask separately from the work to update to Bootstrap 4.

**Comment by TheMegaMatt** [comment]
Was just brainstorming, will do as I get on a PC

Il 22 giu 2018 18:55, "Daniel Roth" <notifications@github.com> ha scritto:

@TheMegaMatt <https://github.com/TheMegaMatt> Sounds like a reasonable ask,
but please file a separate issue so that we can track the ask separately
from the work to update to Bootstrap 4.

—
You are receiving this because you were mentioned.

Reply to this email directly, view it on GitHub
<https://github.com/aspnet/Scaffolding/issues/712#issuecomment-399510004>,
or mute the thread

**Comment by danroth27** [comment]
Splitting off the work to update the identity scaffolder: https://github.com/aspnet/Scaffolding/issues/849

**Comment by ryanbrandenburg** [comment]
Can I get a bit more guidance here? How do we intend to separate the bootstrap4 stuff from the bootstrap3 stuff? Do we have a list of the items which we need to BS4-ify? How can I most easily test these bad boys out to make sure they look good while I'm working on them? This might be a meeting I should have with the Scaffolding folks, but if we also answer some of these questions here we can preserve them for others.

**Comment by ryanbrandenburg** [comment]
@danroth27 do you know the answers to any of my above questions? If not where might I start looking for them?

**Comment by danroth27** [comment]
> How do we intend to separate the bootstrap4 stuff from the bootstrap3 stuff? 

I believe the agreement is that you just need to worry about providing the BS4 content. The scaffolding folks will work out how to integrate it with the existing scaffolders.

> Do we have a list of the items which we need to BS4-ify? 

This issue tracks BS4-ify the Razor Pages with EF Core (CRUD) content and the MVC with EF Core (CRUD) content.

> How can I most easily test these bad boys out to make sure they look good while I'm working on them? 

@mkArtakMSFT Since we're just providing content would it make sense to implement the content as a sample app?

**Comment by ryanbrandenburg** [comment]
I have the content changes @danroth27 described in a branch [here](https://github.com/aspnet/Scaffolding/tree/rybrande/BS4). That's not actually all this issue tracks now that I'm thinking about it though, we also, at minimum, need to update the _Layout.cshtml to match our new layout (though that should just be a copy-paste). @danroth27 are there any other hidden items I'm missing?

**Comment by ryanbrandenburg** [comment]
I updated that branch with out new layout, adjusted to meet the existing standards where I thought appropriate. It does NOT contain an updated site.css (required for the layout to look good) because the only one I saw was in the Identity folder and I didn't want to cause conflicts for @javiercn when he merged his Identity Scaffolding stuff.

I don't see any further work to do here, so if there's something I missed let me know.
